### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by senseihimanshu**

### CODE REVIEW

File: client/src/app/timesheet/modal/modal.component.ts [hunk 1]
```diff
@@ -81,50 +80,47 @@ export class TimesheetModal implements OnInit {
...
+            projectId: project.projectId,
+            projectName: project.projectName,
+            projectManager: project.projectManager,
+            clientName: project.clientName
+          };
+        });
+
+        if(this.data && this.data.timesheetId){
+              this.modalType = 'update';
+              console.log(this.data.timesheetId);
```
Comment (Deepakyadav19) [inline client/src/app/timesheet/modal/modal.component.ts]: Please Remove console logs

File: client/src/app/timesheet/modal/modal.component.ts [hunk 2]
```diff
@@ -201,23 +209,24 @@ export class TimesheetModal implements OnInit {
   }
 
   handleProjectData(project: any) {
+    console.log(project);
```
Comment (Deepakyadav19) [inline client/src/app/timesheet/modal/modal.component.ts]: Please remove console logs

File: client/src/app/timesheet/timesheet-week/timesheet-week.component.ts [hunk 3]
```diff
@@ -123,60 +121,78 @@ export class TimesheetWeekComponent {
...
-   
-      this.timesheet = res.payload.data.timesheet;
-      this.dataSize = res.payload.data.result.dataSize;
-    });
+    this.empObjId = jsonDecoder(localStorage.getItem("Authorization"))._id;
+
+    if(this.role === "admin"){
+      this.timesheetService.getTimesheet({page: this.page.toString(), limit: this.limit.toString(), sort: JSON.stringify(this.sortAccordingTo), criteria: JSON.stringify({}), columns: JSON.stringify({})}).subscribe((res) => {
+        console.log(res, 'Inside Admin Role');
+        this.timesheet = res.payload.data.result;
```
Comment (Deepakyadav19) [inline client/src/app/timesheet/timesheet-week/timesheet-week.component.ts]: Please remove console log

File: client/src/app/timesheet/timesheet-week/timesheet-week.component.ts [hunk 4]
```diff
@@ -123,60 +121,78 @@ export class TimesheetWeekComponent {
...
+        page: this.page.toString(),
+        limit: this.limit.toString(),
+        criteria: JSON.stringify({ empObjId: this.empObjId,  startDate: {$gte: date} }),
+        columns: JSON.stringify({}),
+        sort: JSON.stringify(this.sortAccordingTo)
+      })
       .subscribe(res => {
-        this.timesheet = res.payload.data.filteredTimesheets;
+        console.log(res);
+        this.timesheet = res.payload.data.result;
```
Comment (Deepakyadav19) [inline client/src/app/timesheet/timesheet-week/timesheet-week.component.ts]: please remove console log

File: server/controllers/projectManager.js [hunk 5]
```diff
@@ -1,40 +1,51 @@
...
+              projectManager: 1,
+              projectName: 1,
+              clientName: 1
+            }
+          )]
+        })
+      );
+      
+
+      console.log(projects, projectObjIds, 'Projects');
```
Comment (Deepakyadav19) [inline server/controllers/projectManager.js]: please remove console logs

File: client/src/app/admindashboard/admindashboard.component.ts [hunk 6]
```diff
@@ -64,6 +64,7 @@ export class AdmindashboardComponent implements OnInit {
   }
 
   deleteEmployee(empId: string) {
+    //console.log(empId, "EMPLOYEE ID");
```
Comment (KritikaSachdeva) [inline client/src/app/admindashboard/admindashboard.component.ts]: remove commented code

File: client/src/app/dashboard/dashboard.component.ts [hunk 7]
```diff
@@ -1,74 +1,132 @@
...
-  templateUrl: './dashboard.component.html',
-  styleUrls: ['./dashboard.component.scss']
+  selector: "app-dashboard",
+  templateUrl: "./dashboard.component.html",
+  styleUrls: ["./dashboard.component.scss"]
 })
 export class DashboardComponent implements OnInit, OnChanges {
+  dashboard: string = "C Level Dashboard";
+  graphicaldata: Boolean = true;
+  menus: any = [
```
Comment (KritikaSachdeva) [inline client/src/app/dashboard/dashboard.component.ts]: already included from sidebar, so can be removed from here

File: client/src/app/dashboard/dashboard.component.ts [hunk 8]
```diff
@@ -1,74 +1,132 @@
...
+        }
+      ]
+    }
+  ];
+
+  loading = false;
+  //  users: User[] = [];
+
+  constructor(
+    private httpService: HttpClient,
```
Comment (KritikaSachdeva) [inline client/src/app/dashboard/dashboard.component.ts]: remove till here

File: client/src/app/dashboard/dashboard.component.ts [hunk 9]
```diff
@@ -1,74 +1,132 @@
...
-  projectpieChartLabels =  ['COMPLETED','DISCARDED','IN-PROGRESS'];
-  
+    private employeeService: EmployeeService
+  ) {}
+  pieChartOptions = {
+    responsive: true
+  };
+  // CHART CLICK EVENT.
+  onChartClick(event) {
+    //console.log(event);
```
Comment (KritikaSachdeva) [inline client/src/app/dashboard/dashboard.component.ts]: remove!

File: client/src/app/dashboard/dashboard.component.ts [hunk 10]
```diff
@@ -1,74 +1,132 @@
...
-    ngOnInit() {
-      this.clevelDataProjects(this.graphicaldata);
-      this.clevelDataTimesheets(this.graphicaldata);
-        this.loading = true;
-       
+  timesheetpieChartData: any = [
+    {
+      data: []
     }
+<<<<<<< HEAD
```
Comment (KritikaSachdeva) [inline client/src/app/dashboard/dashboard.component.ts]: ??

File: client/src/app/dashboard/dashboard.component.ts [hunk 11]
```diff
@@ -82,7 +140,7 @@ onChartClick(event) {
  }
 
    ngOnChanges(){}
+>>>>>>> 5b30508c2a5c420fc6cde5231eaa773b96c357fc
```
Comment (KritikaSachdeva) [inline client/src/app/dashboard/dashboard.component.ts]: ??

File: client/src/app/login/login.component.ts [hunk 12]
```diff
@@ -82,40 +82,47 @@ export class LoginComponent implements OnInit {
       password: this.password.nativeElement.value
     };
 
-    this.loginService.login(userObj).subscribe((res) => {
+    this.loginService.login(userObj).subscribe(
+      res => {
+        //console.log(res);
```
Comment (KritikaSachdeva) [inline client/src/app/login/login.component.ts]: !

File: client/src/app/login/login.component.ts [hunk 13]
```diff
@@ -82,40 +82,47 @@ export class LoginComponent implements OnInit {
...
-      if (res != null) {
-        window.localStorage.setItem("Authorization", `Bearer ${res.payload.data['x-auth-token']}`);
+        if (res != null) {
+          window.localStorage.setItem(
+            "Authorization",
+            `Bearer ${res.payload.data["x-auth-token"]}`
+          );
 
-        const token = localStorage.getItem("Authorization");
+          const token = localStorage.getItem("Authorization");
```
Comment (KritikaSachdeva) [inline client/src/app/login/login.component.ts]: not required

File: client/src/app/main/employee-form/employee-form.component.ts [hunk 14]
```diff
@@ -16,39 +16,89 @@ export class EmployeeFormComponent implements OnInit {
   employee: any;
   message: string;
   employeeForm: any;
+  dashboard: string = "Admin Dashboard";
```
Comment (KritikaSachdeva) [inline client/src/app/main/employee-form/employee-form.component.ts]: not required

File: client/src/app/main/employee-form/employee-form.component.ts [hunk 15]
```diff
@@ -16,39 +16,89 @@ export class EmployeeFormComponent implements OnInit {
...
-    private _service :SendHttpRequestService
+    private _service: SendHttpRequestService
   ) {
     this.employeeForm = this.formBuilder.group({
-      name: ['', Validators.required,Validators.minLength(2)],
-     
+      name: ["", Validators.required, Validators.minLength(2)]
     });
   }
+  menus: any = [
```
Comment (KritikaSachdeva) [inline client/src/app/main/employee-form/employee-form.component.ts]: not required

File: client/src/app/myprofile/myprofile.component.ts [hunk 16]
```diff
@@ -6,13 +6,159 @@ import { Component, OnInit } from '@angular/core';
...
   role: any;
+
+  constructor(
+    private _service: EmployeeService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit(): any {
+    const token = localStorage.getItem("Authorization");
```
Comment (KritikaSachdeva) [inline client/src/app/myprofile/myprofile.component.ts]: not required

File: client/src/app/myprofile/myprofile.component.ts [hunk 17]
```diff
@@ -22,6 +168,19 @@ export class MyprofileComponent implements OnInit {
    return this._service.getEmployee(decodeToken.empId).subscribe((response: any) => {  
       return (this.employee = response.payload.data.employee);
       });
-}
+  }
 
+  jsonDecoder = token => {
```
Comment (KritikaSachdeva) [inline client/src/app/myprofile/myprofile.component.ts]: not required

File: client/src/app/project/project.component.ts [hunk 18]
```diff
@@ -13,20 +13,62 @@ import { isRegExp } from 'util';
 })
 export class ProjectComponent implements OnInit {
   name = "Angular";
-  page :number = 1;
-  lastPage:number;
+  page: number = 1;
+  lastPage: number;
   pageSize = 10;
   items = [];
+  menus: IMenu[] = [
```
Comment (KritikaSachdeva) [inline client/src/app/project/project.component.ts]: not required

File: client/src/app/services.service.ts [hunk 19]
```diff
@@ -1,48 +1,57 @@
...
 @Injectable({
-  providedIn: 'root'
+  providedIn: "root"
 })
 export class ServicesService {
+  constructor(private http: HttpClient) {}
 
-  constructor(private http: HttpClient) { }
- 
   private log(message: string) {
```
Comment (KritikaSachdeva) [inline client/src/app/services.service.ts:21]: function can be removed

File: client/src/app/services.service.ts [hunk 20]
```diff
@@ -1,48 +1,57 @@
...
+    localStorage.getItem("Authorization")
+  );
 
-  jsonDecoder = (token) => {
-    var base64Url = token.split('.')[1];
-    var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
-    var jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c) {
-      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
-    }).join(''));
+  jsonDecoder = token => {
```
Comment (KritikaSachdeva) [inline client/src/app/services.service.ts:29]: use this function from utils

File: client/src/app/services/send-http-request.service.ts [hunk 21]
```diff
@@ -1,57 +1,55 @@
...
-
- 
   // //Decode JWT and return the Payload in JSON Format
-  jsonDecoder = (token) => {
-    var base64Url = token.split('.')[1];
-    var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
-    var jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c) {
-      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
-    }).join(''));
+  jsonDecoder = token => {
```
Comment (KritikaSachdeva) [inline client/src/app/services/send-http-request.service.ts:14]: use this function from utils

File: client/src/app/timesheet/modal/modal.component.html [hunk 22]
```diff
@@ -114,13 +112,13 @@ <h4 class="modal-title" id="modal-basic-title">{{ (modalType === "update")? 'Upd
                     ></mat-checkbox>
                   </span>
                 </td>
-                <td>
+                <!-- <td>
```
Comment (KritikaSachdeva) [inline client/src/app/timesheet/modal/modal.component.html:115]: remove commented code

File: client/src/app/timesheet/modal/modal.component.ts [hunk 23]
```diff
@@ -136,19 +132,32 @@ export class TimesheetModal implements OnInit {
...
     const dataToSave = this.formatData(timesheetData);
 
     this.timesheetService
       .createTimesheet(dataToSave)
-      .subscribe((response: any) => {
+      .subscribe((response: IResponse) => {
         Swal.fire(response.payload.message);
       });
+
+      console.log(timesheetData, 'Shivani Bansal');
```
Comment (KritikaSachdeva) [inline client/src/app/timesheet/modal/modal.component.ts]: ???

File: client/src/app/timesheet/timesheet.component.ts [hunk 24]
```diff
@@ -43,13 +42,14 @@ export class TimesheetComponent implements OnInit {
   page = 1;
   pageSize = 10;
   items = [];
+  dashboard: string = "Admin Dashboard";
```
Comment (KritikaSachdeva) [inline client/src/app/timesheet/timesheet.component.ts]: not required

**Review by KritikaSachdeva — CHANGES_REQUESTED** [review]
please make the suggested changes

**Comment by senseihimanshu** [comment]
All suggested changes are made! @KritikaSachdeva

**Review by KritikaSachdeva — APPROVED** [review]

**Pull request merged**